site/source/docs/compiling/WebAssembly.rst

```diff
@@ -22,17 +22,22 @@ WebAssembly is emitted by default, without the need for any special flags.
 
   emcc [..args..] -s WASM=0
 
-.. note:: Emscripten's WebAssembly support depends on `Binaryen <https://github.com/WebAssembly/binaryen>`_, which will be automatically fetched and built for you (you may see logging about that, the first time you compile to WebAssembly).
-.. note:: The ``WASM``, ``BINARYEN*``, etc. options only matter when compiling to your final executable. In other words, the same .o files are used for both asm.js and WebAssembly. Only when linking them and compiling to asm.js or WebAssembly do you need to specify WebAssembly if you want that. That means that it is easy to build your project to both asm.js and WebAssembly.
+.. note:: Emscripten's WebAssembly support depends on `Binaryen <https://github.com/WebAssembly/binaryen>`_, which is provided by the emsdk (if you don't use the emsdk, you need to build it and set it up in your ``.emscripten`` file).
+.. note:: The ``WASM``, ``BINARYEN*``, etc. options only matter when compiling to your final executable. In other words, the same .o files are used for both asm.js and WebAssembly. Only when linking them and compiling to asm.js or WebAssembly do you need to specify WebAssembly if you want that. That means that it is easy to build your project to both asm.js and WebAssembly. (The only exception is the wasm backend's ``WASM_OBJECT_FILES`` flag which determines whether object files contain wasm or LLVM IR.)
 
 Backends
 --------
 
-Emscripten can currently (April 2019) use 2 backends to generate WebAssembly: **fastcomp** (the asm.js backend, together with asm2wasm) and the **upstream LLVM wasm backend**.
+Emscripten can currently (July 2019) use 2 backends to generate WebAssembly: **fastcomp** (the asm.js backend, together with asm2wasm) and the **upstream LLVM wasm backend**.
 
-Fastcomp is currently the default, but we hope to switch the default soon to the upstream backend.
+Fastcomp is currently the default, but we hope to `switch the default soon to the upstream backend <https://v8.dev/blog/emscripten-llvm-wasm>`_.
 
-To use fastcomp, just use the emsdk normally to get ``latest``. For the upstream backend, you can use ``latest-upstream`` for now on Linux, until we finish setting up builders, or you can set LLVM in the ``.emscripten`` file to point to a build you make of very recent LLVM (preferably from git/svn master).
+To use fastcomp, just use the emsdk normally to get ``latest``. For the upstream backend, use ``latest-upstream`` (or, if you are not using the emsdk, you can set LLVM in the ``.emscripten`` file to point to a build you make of very recent LLVM - preferably from git/svn master).
+
+There are some differences you may notice between the two backends, if you upgrade from fastcomp to upstream:
+
+* The wasm backend is strict about linking files with different features sets - for example, if one file was built with atomics but another was not, it will error at link time. This prevents possible bugs, but does may mean you need to make some build system fixes.
```

Comment:
The only strictness is currently for atomics (not other features), and only when doing otherwise would certainly be thread-unsafe (so one being built with pthreads and the other not is insufficient to trigger the error).